lib/rhc/deployment_helpers.rb

```diff
@@ -36,76 +38,104 @@ def deploy_artifact(rest_app, artifact, hot_deploy, force_clean_build)
           end
           raise IncompatibleDeploymentTypeException
         else
-          deploy_git_ref(rest_app, artifact, hot_deploy, force_clean_build)
+          deploy_git_ref(rest_app, artifact, options.hot_deploy, options.force_clean_build, ssh_executable, !!options.ssh)
         end
       end
 
-      def deploy_git_ref(rest_app, ref, hot_deploy, force_clean_build)
+      def deploy_git_ref(rest_app, ref, hot_deploy, force_clean_build, ssh_executable, use_custom_ssh)
```

Comment:
I think `options` is available here (and also in `deploy_local_file` and `deploy_file_from_url`), so you could use `options.ssh` directly instead of the `use_custom_ssh` parameter.